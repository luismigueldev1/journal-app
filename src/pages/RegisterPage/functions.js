import validator from "validator";
import { setErrorAction, removeErrorAction } from "../../actions/uiActions";

export const isFormValid = (dispatch, values) => {
  const { name, email, password, password2 } = values;
  if (name.trim().length === 0) {
    dispatch(setErrorAction("Name is required"));
    return false;
  } else if (!validator.isEmail(email)) {
    dispatch(setErrorAction("Email is not valid"));
    return false;
  } else if (password !== password2 || password.length < 5) {
    dispatch(
      setErrorAction(
        "Password should be at least 6 characters and match each others"
      )
    );

    return false;
  }
  dispatch(removeErrorAction());
  return true;
};
