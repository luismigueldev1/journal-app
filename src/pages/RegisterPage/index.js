import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { useForm } from "../../hooks/useForm";
import { isFormValid } from "./functions";
import { registerWithEmailAndPasswordAction } from "../../actions/authActions";
import { removeErrorAction } from "../../actions/uiActions";

export default function RegisterPage() {
  const [values, handleInputChange] = useForm({
    name: "Luis Miguel",
    email: "luismiguel@gmail.com",
    password: "123456",
    password2: "123456",
  });

  const dispatch = useDispatch();
  const { loading, error } = useSelector((state) => state.ui);

  const handleRegister = (event) => {
    event.preventDefault();
    if (isFormValid(dispatch, values)) {
      dispatch(
        registerWithEmailAndPasswordAction(
          values.email,
          values.password,
          values.name
        )
      );
    }
  };

  useEffect(() => {
    return () => {
      dispatch(removeErrorAction());
    };
  }, [dispatch]);

  return (
    <div className="auth__container">
      <h3 className="text--center">Register Page</h3>

      {error && (
        <div className="alert alert-danger">
          <p className="alert-title ">{error}</p>
        </div>
      )}

      <form
        onSubmit={handleRegister}
        className="animate__animated animate__fadeIn animate__faster"
      >
        <div className="form__control">
          <label htmlFor="name">Name</label>
          <input
            type="text"
            name="name"
            value={values.name}
            onChange={handleInputChange}
          />
        </div>
        <div className="form__control">
          <label htmlFor="email">Email</label>
          <input
            type="text"
            name="email"
            value={values.email}
            onChange={handleInputChange}
          />
        </div>
        <div className="form__control">
          <label htmlFor="password">Password</label>
          <input
            type="password"
            name="password"
            value={values.password}
            onChange={handleInputChange}
          />
        </div>
        <div className="form__control">
          <label htmlFor="password2">Repeat password</label>
          <input
            type="password"
            name="password2"
            value={values.password2}
            onChange={handleInputChange}
          />
        </div>
        <div className="form__control">
          <button type="submit" className="pointer" disabled={loading}>
            {loading ? "Loading..." : "Register"}
          </button>
        </div>

        <hr />
        <div className="google__login pointer">
          <p>
            <b>Login with social:</b>
          </p>
          <button className="pointer mt-1">
            <picture>
              <img src="/assets/google-logo.png" alt="Google Login" />
            </picture>

            <p>
              <b>Sign in with google</b>
            </p>
          </button>
        </div>
        <div className="create__account">
          <Link to="/auth/login">Already register? Login</Link>
        </div>
      </form>
    </div>
  );
}
