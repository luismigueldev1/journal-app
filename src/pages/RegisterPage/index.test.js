import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import RegisterPage from ".";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  auth: {},
  ui: { loading: false, error: null },
  notes: [],
};
let store = mockStore(initialState);
//store.dispatch = jest.fn();

describe("Pruebas <RegisterPage/>", () => {
  beforeEach(() => {
    store = mockStore(initialState);
    jest.clearAllMocks();
  });

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <RegisterPage />
      </MemoryRouter>
    </Provider>
  );
  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de hacer el dispatch de la accion respectiva ()", () => {
    const emailField = wrapper.find('input[name="email"]');

    emailField.simulate("change", {
      target: {
        value: "",
        name: "email",
      },
    });
    wrapper.find("form").simulate("submit");

    const actions = store.getActions();
  });
});
