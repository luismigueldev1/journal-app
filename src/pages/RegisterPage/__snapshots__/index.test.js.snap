// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <RegisterPage/> Debe de mostrarse correctamente 1`] = `
<div
  className="auth__container"
>
  <h3
    className="text--center"
  >
    Register Page
  </h3>
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <div
      className="form__control"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        name="name"
        onChange={[Function]}
        type="text"
        value="Luis Miguel"
      />
    </div>
    <div
      className="form__control"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value="luismiguel@gmail.com"
      />
    </div>
    <div
      className="form__control"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value="123456"
      />
    </div>
    <div
      className="form__control"
    >
      <label
        htmlFor="password2"
      >
        Repeat password
      </label>
      <input
        name="password2"
        onChange={[Function]}
        type="password"
        value="123456"
      />
    </div>
    <div
      className="form__control"
    >
      <button
        className="pointer"
        disabled={false}
        type="submit"
      >
        Register
      </button>
    </div>
    <hr />
    <div
      className="google__login pointer"
    >
      <p>
        <b>
          Login with social:
        </b>
      </p>
      <button
        className="pointer mt-1"
        onClick={[Function]}
      >
        <picture>
          <img
            alt="Google Login"
            src="/assets/google-logo.png"
          />
        </picture>
        <p>
          <b>
            Sign in with google
          </b>
        </p>
      </button>
    </div>
    <div
      className="create__account"
    >
      <a
        href="/auth/login"
        onClick={[Function]}
      >
        Already register? Login
      </a>
    </div>
  </form>
</div>
`;
